estate/models/estate_property_type.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api
+
+
+class EstatePropertyType(models.Model):
+    _name = "estate.property.type"
+    _description = "Estate Property Type"
+    _order = "sequence"
+
+    name = fields.Char(required=True)
+    property_ids = fields.One2many("estate.property","property_type_id")
+    sequence = fields.Integer(default=1)
+    offer_ids = fields.One2many("estate.property.offer","property_type_id")
+    offer_count = fields.Integer(compute="_compute_offer_count",store=True)
+
+    _sql_constraints = [
+        ('check_unique_name', 'unique(name)',
+         'The type must be unique')
+
+    ]
+
+    @api.depends("offer_ids")
+    def _compute_offer_count(self):
+        for property_type in self:
+            property_type.offer_count = 0
+            for offer in property_type.offer_ids:
+                property_type.offer_count += 1
+
```

Comment:
`property_id.offer_ids` is a recordset, so you can get its size with `len(property_type.offer_ids)` which would be less error prone and more pythonesque